inc/Base/DataHandler.php

```diff
@@ -97,13 +97,14 @@ public static function generate_rss_feed( $category, $limit = 50 ) {
 			  <smly:old_price>' . esc_attr( $price ) . '</smly:old_price>
 			  <smly:discount>-' . esc_attr( $discount ) . '%</smly:discount>';
 			}
-
+			// Remove image if inserted to description.
+			$description = preg_replace( '/\[.*caption\]/', '', ( $prod->get_description() ) );
 			$items[] = '<item>
 			  <title>' . esc_attr( $prod->get_title() ) . '</title>
 			  <link>' . esc_url( $url ) . '</link>
 			  <guid isPermaLink="True">' . esc_url( $url ) . '</guid>
 			  <pubDate>' . date( 'D, d M Y H:i:s', $create_time ) . '</pubDate>
-			  <description>' . htmlentities( $prod->get_description() ) . '</description>
+			  <description><![CDATA[' . esc_attr( $description ) . ']]></description>
```

Comment:
No need to add `esc_attr` when using `<![CDATA[...]]>` tags.

Reply:
So if using `do_shortcode ` and not escaping it. all content between` [caption] `will be entered directly.  Dont know how ower RSS renderer handles output from `do_shortcode`. Might get weird if not showing only text I think.
![image](https://user-images.githubusercontent.com/22918359/52163613-09800400-26ed-11e9-9b2d-b9bcc49f17cf.png)
alert is written between caption tags in product editor.

If using `esc_attr` then all tags are converted to entities, but not ÄÖÜ etc. So xml wont crash.
![image](https://user-images.githubusercontent.com/22918359/52163636-7c897a80-26ed-11e9-8551-4b62f40f7264.png)
Using `htmlentities `converts all. Without  `<![CDATA[]]>` xml parser crashes.
![image](https://user-images.githubusercontent.com/22918359/52163656-beb2bc00-26ed-11e9-979b-dce4ff5656a4.png)

So the question is if XML-parser on our end handles `<![CDATA[]]>` tag and can handle entities between it. There is also problem with` nbsp`-entity. (Generated after image in description) Crashes without  `CDATA `tags.

If it does and supports parsing entities `<![CDATA[]]>` should be added to description field.  And in that case htmlentities can be used like before. Maybe add it to title field also in that case.

I can change it in any way.

For a quick fix I can remove `[caption]` and ` nbsp` - entities before escaping with `esc_attr` and obmit  `<![CDATA[]]>` tag. This leaves ÖÜÕÄ untouched and can render xml with previous structure. So estonian stores can have products with ÜÖÄÜ in their names or description.

Reply:
Copy that, will modify accordingly